File: Programmers/Python/Level1/Practice_test.py
def solution(answers):
    answer = []
    score = [0 for i in range(3)]
    first_way = 1
    second_way = 1
    
    for i in range(len(answers)):
        # 1번 수포자
        if first_way == answers[i]:
            score[0] += 1
            
        first_way += 1
        if first_way > 5: 
            first_way = 1
        
        
        # 2번 수포자
        if i % 2 == 0:
            if answers[i] == 2:
                score[1] += 1
        else:
            if answers[i] == second_way:
                score[1] += 1
                
            second_way += 1
            if second_way == 2:
                second_way += 1
            if second_way > 5: 
                second_way = 1
        
        
        # 3번 수포자
        if i % 10 == 0 or i % 10 == 1:
            if answers[i] == 3:
                score[2] += 1
        elif i % 10 == 2 or i % 10 == 3:
            if answers[i] == 1:
                score[2] += 1
        elif i % 10 == 4 or i % 10 == 5:
            if answers[i] == 2:
                score[2] += 1
        elif i % 10 == 6 or i % 10 == 7:
            if answers[i] == 4:
                score[2] += 1
        elif i % 10 == 8 or i % 10 == 9:
            if answers[i] == 5:
                score[2] += 1
    
    
    top_rank = max(score)
    for i in range(len(score)):
        if top_rank == score[i]:
            answer.append(i + 1)
            
    return answer

File: Programmers/C++/Level1/Practice_test.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> scores;
    
    int first_selection = 1;
    int first_score = 0;
    
    int second_selection = 1;
    int second_score = 0;
    
    int third_score = 0;
    
    for(int i = 0; i < answers.size(); i++)
    {
        // 1번 수포자
        if(answers[i] == first_selection) first_score++;
        first_selection++;
        if(first_selection > 5) first_selection = 1;
        
        // 2번 수포자
        if(i % 2 == 0)
        {
            if(answers[i] == 2) second_score++;
        }
        else
        {
            if(answers[i] == second_selection) second_score++;
            second_selection++;
            if(second_selection == 2) second_selection++;
            if(second_selection > 5) second_selection = 1;
        }
        
        // 3번 수포자
        switch(i % 10)
        {
            case 0:
            case 1:
                if(answers[i] == 3) third_score++;
                break;
            case 2:
            case 3:
                if(answers[i] == 1) third_score++;
                break;
            case 4:
            case 5:
                if(answers[i] == 2) third_score++;
                break;
            case 6:
            case 7:
                if(answers[i] == 4) third_score++;
                break;
            case 8:
            case 9:
                if(answers[i] == 5) third_score++;
                break;
        }
    }
    scores.push_back(first_score);
    scores.push_back(second_score);
    scores.push_back(third_score);
    
    int top_rank = *max_element(scores.begin(), scores.end());
    for(int i = 0; i < scores.size(); i++)
    {
        if(top_rank == scores[i]) answer.push_back(i+1);
    }
    
    return answer;
}